### PR TITLE
Adding compat subpackage for kube-proxy

### PR DESCRIPTION
**Pull request by Aditevil**
kube-proxy deployments assume the binary sits at
/usr/local/bin/kube-proxy (see
https://github.com/kubernetes/kubernetes/blob/a35bca903e24ea3cdbecb743c001b4c8f4db9e9c/cmd/kubeadm/app/phases/addons/proxy/manifests.go#L79). This adds a compat package based off of kube-proxy-default (which installs the binary at /usr/bin/kube-proxy) to add a symlink to /usr/local/bin.

Kube-proxy pod expects the binary at` /usr/local/bin`.

logs from kube-proxy-pod:
  `Normal   Created    13s (x3 over 27s)  kubelet            Created container kube-proxy Warning  Failed     13s (x3 over 26s)  kubelet            Error: failed to create containerd task: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/usr/local/bin/kube-proxy": stat /usr/local/bin/kube-proxy: no such file or directory: unknown`

### CODE REVIEW

**Comment by Aditvil-Dev** [comment]
https://github.com/wolfi-dev/advisories/pull/8531

**Review by developer-guy — APPROVED** [review]

**Review by EyeCantCU — CHANGES_REQUESTED** [review]
Why do we need this? Can we please add something to the PR description? Is having this in the local path not enough? We already have a `-default` package for this that also creates a link

File: kubernetes-1.31.yaml [hunk 1]
```diff
@@ -187,6 +187,19 @@ subpackages:
           mkdir -p ${{targets.subpkgdir}}/usr/bin
           ln -s ${{range.key}}-${{vars.kubernetes-version}} ${{targets.subpkgdir}}/usr/bin/${{range.key}}
 
+  - name: kube-proxy-${{vars.kubernetes-version}}-default-compat
```
Comment (Dentrax) [inline kubernetes-1.31.yaml:190]: Is this compat package needs to be versioned?

**Comment by Aditvil-Dev** [comment]
logs from kube-proxy-pod: 
`  Normal   Created    13s (x3 over 27s)  kubelet            Created container kube-proxy
  Warning  Failed     13s (x3 over 26s)  kubelet            Error: failed to create containerd task: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/usr/local/bin/kube-proxy": stat /usr/local/bin/kube-proxy: no such file or directory: unknown`

File: kubernetes-1.31.yaml (hunk 1 above)
Comment (Aditvil-Dev) [reply kubernetes-1.31.yaml:190]: Following the norm for has been done for `-default` packages

**Comment by EyeCantCU** [comment]
Looks good, but please update the PR description explaining why the change was made

**Review by EyeCantCU — APPROVED** [review]

**Pull request merged**